src/main/java/com/google/firebase/messaging/Aps.java

```diff
@@ -137,6 +146,41 @@ public Builder setContentAvailable(boolean contentAvailable) {
       return this;
     }
 
+    /**
+     * Specifies whether to set the {@code mutable-content} property on the message, so the
+     * clients can modify the notification via app extensions.
+     *
+     * @param mutableContent True to make the content mutable via app extensions.
+     * @return This builder.
+     */
+    public Builder setMutableContent(boolean mutableContent) {
+      this.mutableContent = mutableContent;
+      return this;
+    }
+
+    /**
+     * Puts a custom key-value pair to the aps dictionary.
+     *
+     * @param key A non-null key.
+     * @param value A non-null, json-serializable value.
+     * @return This builder.
+     */
+    public Builder putCustomData(@NonNull String key, @NonNull Object value) {
+      this.customData.put(key, value);
```

Comment:
Should we assert here that `mutableContent` is set to true? Please correct me if I am misunderstanding how this works.

Reply:
Well in that case it has to be set as `mutable-content: 1`. Since we are exposing a typed setter to configure this (`setMutabeContent()`), my preference is to not validate it here. And if a developer tries to set some nonsense value, they will get a backend error.